Apromore-Database/src/main/java/org/apromore/dao/model/DataType.java

```diff
@@ -0,0 +1,51 @@
+/*-
+ * #%L
+ * This file is part of "Apromore Enterprise Edition".
+ * %%
+ * Copyright (C) 2019 - 2022 Apromore Pty Ltd. All Rights Reserved.
+ * %%
+ * NOTICE:  All information contained herein is, and remains the
+ * property of Apromore Pty Ltd and its suppliers, if any.
+ * The intellectual and technical concepts contained herein are
+ * proprietary to Apromore Pty Ltd and its suppliers and may
+ * be covered by U.S. and Foreign Patents, patents in process,
+ * and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this
+ * material is strictly forbidden unless prior written permission
+ * is obtained from Apromore Pty Ltd.
+ * #L%
+ */
+
+package org.apromore.dao.model;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import java.util.Locale;
+
+/**
+ * Each column must have a specified data_type from the following list:
+ * "double" (double-precision floating-point number)
+ * "int" (integer number)
+ * "string" (text)
+ * "date" (timestamp)
+ * These values were chosen to match https://www.xes-standard.org/_media/xes/xesstandarddefinition-2.0.pdf §2.2,
+ * which also includes some we’re not using: “boolean”, “id”, “list”.
+ */
+@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
+public enum DataType {
+    STRING,
+    DATE,
+    DOUBLE,
+    INT;
+
```

Comment:
It looks like this DataType is meant to cover the most common primitive data, right? Following this line, I wonder why long is not included here. For example, timestamp is all long values and very common in our data, not sure if the intention here is to cover it with String or Double, but these two are not suitable (String is unchecked type, and Double is lost precision).

Reply:
These values are taken directly from https://apromore.atlassian.net/wiki/spaces/AP/pages/2143748188/Apromore-Log-Meta+header+format

Reply:
I've been pondering this as well. should we expand the DataTypes to include others such as long, timestamp? I briefly discussed this with Marlon as well, and we're going to do the same in the converters as well (which currently only supports `int`, `string`, `date` and `float`.

Given that when we convert these basic types to the Hive schema types we will be translating
`int` -> `bitint` 
`float` -> `double`
`string` -> `varchar(255)`
`date` -> `timestamp`

I believe there would be an advantage in providing a richer set of datatypes. We then of course need to update the REST API as well to make use of these

Reply:
I had already intentionally changed `float` to `double` in the DataTypes because of this

Reply:
From our discussion, I understood this DataType is high level and not mapped to specific types (like Java, SQL). But the naming is quite specific: Double, Int, String (this looks like Java, e.g. why calling it "Double"). A better naming is: String = Text (or Literal), Double = Number_Continuous, Int = Number_Discreet.

Reply:
DataType.DATE here should be "DATETIME". There's a missing type for time only. Java has OffsetTime, and MySQL has TIME data type (https://dev.mysql.com/doc/refman/8.0/en/time.html). In CustomCalendar module, it uses time data only for workday (e.g. from 9am to 5pm).